xbmc/XBDateTime.cpp

```diff
@@ -797,123 +584,70 @@ bool CDateTime::SetDate(int year, int month, int day)
 
 bool CDateTime::SetTime(int hour, int minute, int second)
 {
-  // 01.01.1601 00:00:00 is 0 as filetime
-  return SetDateTime(1601, 1, 1, hour, minute, second);
-}
+  m_time = date::sys_seconds(std::chrono::seconds(0)) + std::chrono::hours(hour) +
+           std::chrono::minutes(minute) + std::chrono::seconds(second);
 
-void CDateTime::GetAsSystemTime(KODI::TIME::SystemTime& time) const
-{
-  FileTimeToSystemTime(&m_time, &time);
+  SetValid(true);
+  return true;
 }
 
-#define UNIX_BASE_TIME 116444736000000000LL /* nanoseconds since epoch */
 void CDateTime::GetAsTime(time_t& time) const
 {
-  long long ll = (static_cast<long long>(m_time.highDateTime) << 32) + m_time.lowDateTime;
-  time=(time_t)((ll - UNIX_BASE_TIME) / 10000000);
+  time = date::floor<std::chrono::seconds>(m_time.time_since_epoch()).count();
 }
 
 void CDateTime::GetAsTm(tm& time) const
 {
-  KODI::TIME::SystemTime st;
-  GetAsSystemTime(st);
+  auto dp = date::floor<date::days>(m_time);
 
```

Comment:
It's probably a good idea to first clear the whole `struct tm` with `time = {};` because depending on the implementation `struct tm` may has additional member (e.g. [glibc](https://www.gnu.org/software/libc/manual/html_node/Broken_002ddown-Time.html)) and having them in a random state can be confusing when looking at it in the debugger.